include/ddc/kernels/splines/spline_builder.hpp

```diff
@@ -264,23 +303,29 @@ class SplineBuilder
      * Values of the function must be provided on this domain in order
      * to build a spline representation of the function (cartesian product of 1D interpolation_domain and batch_domain).
      *
+     * @param batched_interpolation_domain The whole domain on which the interpolation points are defined.
+     *
      * @return The domain for the interpolation mesh.
      */
-    batched_interpolation_domain_type batched_interpolation_domain() const noexcept
+    template <class DDom, class = std::enable_if_t<ddc::is_discrete_domain_v<DDom>>>
+    DDom batched_interpolation_domain(DDom const& batched_interpolation_domain) const noexcept
     {
-        return m_batched_interpolation_domain;
+        return batched_interpolation_domain;
     }
 
     /**
      * @brief Get the batch domain.
      *
      * Obtained by removing the dimension of interest from the whole interpolation domain.
      *
+     * @param batched_interpolation_domain The whole domain on which the interpolation points are defined.
+     *
      * @return The batch domain.
      */
-    batch_domain_type batch_domain() const noexcept
+    template <class DDom>
+    batch_domain_type<DDom> batch_domain(DDom const& batched_interpolation_domain) const noexcept
     {
-        return ddc::remove_dims_of(batched_interpolation_domain(), interpolation_domain());
+        return ddc::remove_dims_of(batched_interpolation_domain, interpolation_domain());
```

Comment:
```suggestion
        assert(interpolation_domain() == interpolation_domain_type(batched_interpolation_domain));
        return ddc::remove_dims_of(batched_interpolation_domain, interpolation_domain());
```

I suggest that we add an assertion to check the user is still manipulating a batched_interpolation_domain with the same interpolation_domain used at the construction of the SplineBuilder.